pkg/types/baremetal/validation/platform_test.go

```diff
@@ -10,6 +13,172 @@ import (
+			BootMACAddress: "CA:FE:CA:FE:00:01",
+			BMC: baremetal.BMC{
+				Username: "root",
+				Password: "password",
+				Address:  "ipmi://192.168.111.2",
+			},
+		},
+	}
+}
+
+func (hb *hostBuilder) build() *baremetal.Host {
+	return &hb.Host
+}
+
+func (hb *hostBuilder) Name(value string) *hostBuilder {
+	hb.Host.Name = value
+	return hb
+}
+
+func (hb *hostBuilder) BootMACAddress(value string) *hostBuilder {
+	hb.Host.BootMACAddress = value
+	return hb
+}
+
+func (hb *hostBuilder) BMCAddress(value string) *hostBuilder {
+	hb.Host.BMC.Address = value
+	return hb
+}
+
+func (hb *hostBuilder) BMCUsername(value string) *hostBuilder {
+	hb.Host.BMC.Username = value
+	return hb
+}
+
+func (hb *hostBuilder) BMCPassword(value string) *hostBuilder {
+	hb.Host.BMC.Password = value
+	return hb
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+type platformBuilder struct {
+	baremetal.Platform
+}
+
+func platform() *platformBuilder {
+	return &platformBuilder{
+		baremetal.Platform{
+			APIVIP:                       "192.168.111.2",
+			DNSVIP:                       "192.168.111.3",
+			IngressVIP:                   "192.168.111.4",
+			Hosts:                        []*baremetal.Host{},
+			LibvirtURI:                   "qemu://system",
+			ProvisioningNetworkCIDR:      ipnet.MustParseCIDR("172.22.0.0/24"),
+			ClusterProvisioningIP:        "172.22.0.3",
+			BootstrapProvisioningIP:      "172.22.0.2",
+			ExternalBridge:               "br0",
+			ProvisioningBridge:           "br1",
+			ProvisioningNetworkInterface: "ens3",
+		}}
+}
+
+func (pb *platformBuilder) build() *baremetal.Platform {
+	return &pb.Platform
+}
+
+func (pb *platformBuilder) ProvisioningNetworkCIDR(value string) *platformBuilder {
+	pb.Platform.ProvisioningNetworkCIDR = ipnet.MustParseCIDR(value)
+	return pb
+}
+
+func (pb *platformBuilder) ClusterProvisioningIP(value string) *platformBuilder {
+	pb.Platform.ClusterProvisioningIP = value
+	return pb
+}
+
+func (pb *platformBuilder) BootstrapProvisioningIP(value string) *platformBuilder {
+	pb.Platform.BootstrapProvisioningIP = value
+	return pb
+}
+
+func (pb *platformBuilder) BootstrapOSImage(value string) *platformBuilder {
+	pb.Platform.BootstrapOSImage = value
+	return pb
+}
+
+func (pb *platformBuilder) ClusterOSImage(value string) *platformBuilder {
+	pb.Platform.ClusterOSImage = value
+	return pb
+}
+
+func (pb *platformBuilder) ProvisioningDHCPRange(value string) *platformBuilder {
+	pb.Platform.ProvisioningDHCPRange = value
+	return pb
+}
+
+func (pb *platformBuilder) APIVIP(value string) *platformBuilder {
+	pb.Platform.APIVIP = value
+	return pb
+}
+
+func (pb *platformBuilder) DNSVIP(value string) *platformBuilder {
+	pb.Platform.DNSVIP = value
+	return pb
+}
+
+func (pb *platformBuilder) IngressVIP(value string) *platformBuilder {
+	pb.Platform.IngressVIP = value
+	return pb
+}
+
+func (pb *platformBuilder) Hosts(value ...*baremetal.Host) *platformBuilder {
+	pb.Platform.Hosts = value
```

Comment:
Neat, I didn't realize variadic function parameters could just be assigned to a slice